src/components/labeledInput.tsx

```diff
@@ -0,0 +1,271 @@
+import React, { ReactElement } from 'react'
+import { TextField, Box, Checkbox, FormControlLabel } from '@material-ui/core'
+import { Control, Controller, RegisterOptions } from 'react-hook-form'
+import InputMask from 'react-input-mask'
+import locationPostalCodes from '../data/locationPostalCodes'
+
+interface Errors {
+  [key: string]: {type: string}
+}
+
+type Register = (rules?: RegisterOptions) => any
+
+interface PatternConfig {
+  mask?: string
+  regexp?: RegExp
+  description?: string
+}
+
+interface LabeledInputProps {
+  strongLabel?: string
+  patternConfig?: PatternConfig
+  label: string
+  register: Register
+  required: boolean
+  name: string
+  errors: Errors
+  defaultValue?: string
+}
+
+const getError = (errors: Errors, name: string): string | undefined => {
+  let result
+  if (Object.prototype.hasOwnProperty.call(errors, name)) {
+    const r = errors[name]
+    result = r.type
+  }
+  return result
+}
+
+const isError = (errors: Errors, name: string): boolean => getError(errors, name) !== undefined
+
+export function LabeledInput (props: LabeledInputProps): ReactElement {
+  const { strongLabel, label, register, required, patternConfig = {}, name, errors, defaultValue } = props
+  let helperText: string | undefined
+  // fix error where pattern wouldn't match if input wasn't filled out even if required was set to false
+  if (!required) {
+    patternConfig.regexp = /.?/
+  }
+  if (errors[name]?.type === 'required') {
+    helperText = 'Input is required'
+  } else if (errors[name]?.type === 'pattern') {
+    // Must have a regex pattern description so users can see what's going wrong *
+    helperText = patternConfig.description
+  }
+
+  /* default regex pattern is to accept any input. Otherwise, use input pattern */
+  const textField = (
+    <TextField
+      error={isError(errors, name)}
+      helperText={helperText}
+      fullWidth={patternConfig.mask === undefined}
+      defaultValue={defaultValue}
+      inputRef={register({ required: required, pattern: patternConfig.regexp })}
+      name={name}
+      variant="filled"
+    />
+  )
+
+  /* if there is a mask prop, create masked textfield rather than standard */
+  let input = textField
+  if (patternConfig.mask !== undefined) {
+    input = (
+      <InputMask
+        mask={patternConfig.mask}
+        alwaysShowMask={true}
+        defaultValue={defaultValue}
+      >
+        {textField}
+      </InputMask>
+    )
+  }
+
+  return (
+    <div>
+      <Box display="flex" justifyContent="flex-start">
+        <p><strong>{strongLabel}</strong>{label}</p>
+      </Box>
+      <Box display="flex" justifyContent="flex-start">
+        {input}
+      </Box>
+    </div>
+  )
+}
+
+const id = <A,>(a: A): A => a
```

Comment:
what's this function doing?

Reply:
Re: linter error, definitely looks like that should be a linter error, trying to figure out why I don't see it. Do you see it if you run `npm run lint`? What version of eslint is your vscode using?

Reply:
I reorganized the labeledInput file a little bit so that hopefully it's a little more clear. This is just a default parameter for a LabeledDropdown that only contains an array of strings.

Reply:
What does A and a signify? are they just generic types?

Reply:
Yeah it explicitly doesn't signify anything, so that's why that naming convention is used.

```
  // this function takes an A which can be anything, and returns an A, so we know it can 
  // only possibly return the input of the function.
  const id = <A,>(a: A): A => a

  // return type tells you this can return any string in the universe
  const doNothing = (optionText: string): string => optionText

  // still hints it could return any string in the universe
  const doNothing2: ((x: string) => string) = id

  // any possible number, who knows
  const doNothing3 = (n: number): number => n
  const doNothing4: ((n: number) => number) = id
```